src/coreclr/src/debug/createdump/createdumpwindows.cpp

```diff
@@ -2,28 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 #include "createdump.h"
+#include "psapi.h"
 
 //
 // The Windows create dump code
 //
 bool
-CreateDump(const char* dumpPath, int pid, MINIDUMP_TYPE minidumpType)
+CreateDump(const char* dumpPathTemplate, int pid, const char* dumpType, MINIDUMP_TYPE minidumpType)
 {
     HANDLE hFile = INVALID_HANDLE_VALUE;
     HANDLE hProcess = NULL;
     bool result = false;
 
+    ArrayHolder<char> pszName = new char[MAX_LONGPATH + 1];
+    std::string dumpPath;
+
     hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, pid);
     if (hProcess == NULL)
     {
         fprintf(stderr, "Invalid process id '%d' error %d\n", pid, GetLastError());
         goto exit;
     }
+    if (GetModuleBaseNameA(hProcess, NULL, pszName, MAX_LONGPATH) <= 0)
+    {
+        fprintf(stderr, "Get process name FAILED %d\n", GetLastError());
+        goto exit;
+    }
+    if (!FormatDumpName(dumpPath, dumpPathTemplate, pszName, pid))
+    {
+        goto exit;
+    }
+    printf("Writing %s to file %s\n", dumpType, dumpPath.c_str());
 
-    hFile = CreateFileA(dumpPath, GENERIC_READ | GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    hFile = CreateFileA(dumpPath.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
     if (hFile == INVALID_HANDLE_VALUE)
     {
-        fprintf(stderr, "Invalid dump path '%s' error %d\n", dumpPath, GetLastError());
+        fprintf(stderr, "Invalid dump path '%s' error %d\n", dumpPath.c_str(), GetLastError());
```

Comment:
Where a message could result from bad user input, like here, consider an error string (FormatMessage()) so they don't have to look up the code?

Reply:
Using FormatMessage is a lot of code so I'd rather not for this PR.